apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs

```diff
@@ -208,6 +206,17 @@ public string GetCurrentTraceId()
             }
         }
 
+        /// <summary>
+        /// Sets a <see cref="StackTrace"/> on the current span for the given exception and
+        /// returns false.  This is used for exception handling to ensure proper timing for
+        /// recording the stacktrace.
```

Comment:
It's more that it avoids sometimes losing data in the stack trace that's eventually recorded, but the details are somewhat detailed :)

Did you check that this still behaves the way we want it to?